server/utilitis.go

```diff
@@ -2,31 +2,34 @@ package server
 
 import (
 	"encoding/json"
+	"github.com/rs/zerolog"
 	"net/http"
 )
 
-func writeJsonResponse(w http.ResponseWriter, statusCode int, response any) {
+func writeJsonResponse(w http.ResponseWriter, req *http.Request, log *zerolog.Logger, response any) {
 	if response == nil {
 		panic("writeJsonRespone: empty response data")
 	}
 
 	jsonData, err := json.Marshal(response)
 	if err != nil {
-		ErrorResponse(w, ErrorFailedMarshalJSON, "failed to marshal JSON response", http.StatusInternalServerError)
+		ErrorResponse(w, req, ErrorFailedMarshalJSON, "failed to marshal JSON response", http.StatusInternalServerError, log)
 		return
 	}
 
-	writeRespone(w, http.StatusOK, "application/json", jsonData)
+	writeResponse(w, req, log, http.StatusOK, "application/json", jsonData)
 }
 
-func writeRespone(w http.ResponseWriter, statusCode int, contentType string, responseData []byte) {
+func writeResponse(w http.ResponseWriter, req *http.Request, log *zerolog.Logger, statusCode int, contentType string, responseData []byte) {
 	w.Header().Set("Content-Type", contentType)
 	w.WriteHeader(statusCode)
 
+	log.Debug().Msgf("%d | %s | %s | %s ", statusCode, req.RemoteAddr, req.Method, req.URL)
```

Comment:
Additionally we could add those params in contextual fields and maybe some additional marker like `Str("logger", "http-response")`

Reply:
Done